src/coreclr/src/System.Private.CoreLib/src/System/Math.CoreCLR.cs

```diff
@@ -11,119 +11,176 @@
 ===========================================================*/
 
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace System
 {
     public static partial class Math
     {
+        [SuppressGCTransition]
+        [DllImport(RuntimeHelpers.QCall)]
```

Comment:
These need to have an explicitly specific CDecl calling convention to make it work on Windows x86.

Reply:
Added. In other places, where QCall is used, we have not specified Cdecl. Is it because we have registered the C math functions directly with QCalls in this case?

Reply:
>  Is it because we have registered the C math functions directly with QCalls in this case?

Yup.